libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFWKWebViewEngine/SFWKWebViewDelegate.h

```diff
@@ -0,0 +1,45 @@
+/*
+ SFWKWebViewDelegate.h
+ SalesforceHybridSDK
+
+ Created by Bharath Hariharan on 7/15/16.
+
+ Copyright (c) 2016, salesforce.com, inc. All rights reserved.
+
+ Redistribution and use of this software in source and binary forms, with or without modification,
+ are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this list of conditions
+ and the following disclaimer.
+ * Redistributions in binary form must reproduce the above copyright notice, this list of
+ conditions and the following disclaimer in the documentation and/or other materials provided
+ with the distribution.
+ * Neither the name of salesforce.com, inc. nor the names of its contributors may be used to
+ endorse or promote products derived from this software without specific prior written
+ permission of salesforce.com, inc.
+
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import <WebKit/WebKit.h>
+#import <Cordova/CDVAvailability.h>
+
+@interface SFWKWebViewDelegate : NSObject <WKNavigationDelegate> {
+    __weak NSObject <WKNavigationDelegate> *_delegate;
```

Comment:
I know a lot of this was copied from `CDVUIWebViewDelegate`, but it could use some cleanup/reorganization.  Starting here: it's generally more practical to manage member variables from privately-defined properties these days, so that there are no ambiguities on the memory management side.  If member variables are still used, they're almost exclusively defined in a private `@interface` in the .m file.  This is pretty old-school.